Reduce number of raw types warnings

cbuescher opened the pull request:
Among the ~7000 warnings we currently have, raw type warnings are one of the biggest 
group (aprox. 1700) and one that is the easiest to reduce, I think. This PR is a first attempt
to reduce this number, most of the time by using the unbounded wildcard.

elasticmachine commented:
Pinging @elastic/es-core-infra

cbuescher commented:
As an aside: figuring out why we have so many generic types (and reducing that number) is a different matter I think.

rjernst commented on x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/AuthenticationServiceTests.java:
```diff
@@ -688,6 +688,7 @@ public void testRealmLookupThrowingExceptionRest() throws Exception {
         }
     }
 
+    @SuppressWarnings("unchecked")
```
Why is this suppression still necessary (and the others in this file)?

cbuescher replied:
In this particular case L699 `i.getArguments()` returns an Object[] which needs to be cast to ActionListener<User>. Some of the Mockito stuff in here also requires SupressWarnings but I haven't added it elsewhere to keep this PR focused in raw tapes removals. I also haven't removed some other warnings in the classes touched since I think its makes any clean-up PR like this harder to review.

rjernst reviewed (approved):
LGTM

Commit:
Reduce number of raw types warnings

Among the ~7000 warnings we currently have, raw type warnings are one of the
biggest group (aprox. 1700) and one that is the easiest to reduce. This PR is a
fist attempt to reduce this number, most of the time by using the unbounded
wildcard.

Commit:
Merge branch 'master' into reduce-rawTypes

 Conflicts:
	server/src/test/java/org/elasticsearch/common/geo/BaseGeoParsingTestCase.java
	server/src/test/java/org/elasticsearch/common/geo/builders/AbstractShapeBuilderTestCase.java

cbuescher commented:
@rjernst @jpountz I'm not sure if we want to backport non-issues with changes across many classes to 6.x. Does it make merging back easier or more complicated? Wdyt?

rjernst commented:
I would not backport cleanups like this unless there is a compelling reason to do so.
